titanic.py

```diff
@@ -134,23 +127,15 @@ def run_demo():
 
     sorted_scores = sorted(scores.items(), key=lambda pair: pair[1], reverse=True)
     dataset = dataset[[column[0] for column in sorted_scores[0:keep]] + ["survived"]]
-    features = list(set(dataset.columns).difference(('survived',)))
-    features = sorted(features)
+    features = sorted(list(set(dataset.columns) - {'survived'}))
 
-    # Build a QUBO that maximizes MI between survival and a subset of features
-    bqm = dimod.BinaryQuadraticModel.empty(dimod.BINARY)
+    variables = {feature: -mutual_information(prob(dataset[['survived', feature]].values), 1)
+                 for feature in features}
 
-    # Add biases as (negative) MI with survival for each feature
-    for feature in features:
-        mi = mutual_information(prob(dataset[['survived', feature]].values), 1)
-        bqm.add_variable(feature, -mi)
+    interactions = {(f0, f1): -conditional_mutual_information(prob(dataset[['survived', f0, f1]].values), 1, 2)
+                    for f0, f1 in itertools.permutations(features, 2)}
 
-    # Add interactions as (negative) MI with survival for each set of 2 features
-    for f0, f1 in itertools.combinations(features, 2):
-        cmi_01 = conditional_mutual_information(prob(dataset[['survived', f0, f1]].values), 1, 2)
-        cmi_10 = conditional_mutual_information(prob(dataset[['survived', f1, f0]].values), 1, 2)
-        bqm.add_interaction(f0, f1, -cmi_01)
-        bqm.add_interaction(f1, f0, -cmi_10)
+    bqm = dimod.BinaryQuadraticModel(variables, interactions, 0, dimod.BINARY)
```

Comment:
This will generally be much less memory efficient than building on the BQM directly. For small examples like this it won't make a difference but in general we like to use the more scalable approach.

Reply:
Good point, might be better to go back to the incremental construction, seeing how these examples tend to be used as templates/guides when working up towards bigger problems.

Reply:
Is there a utility function that constructs a BQM from a pair iterables of tuples, ie a lazy version of the constructor

Reply:
There is. You can in-fact do `interactions = ((u, v, bias) for ...)` which (perhaps) gives the best of both

Reply:
Cool! I did not realise you could do that. The docs only mention dicts. (I accidentally  resolved the abve threads)

Reply:
Yeah, a lot of that is getting cleaned up as part of the 0.10.x work.

Reply:
Done